mmpose/datasets/pipelines/top_down_transform.py

```diff
@@ -179,16 +188,24 @@ def __call__(self, results):
         c = results['center']
         s = results['scale']
         r = results['rotation']
-        trans = get_affine_transform(c, s, r, image_size)
-
-        img = cv2.warpAffine(
-            img,
-            trans, (int(image_size[0]), int(image_size[1])),
-            flags=cv2.INTER_LINEAR)
 
-        for i in range(results['ann_info']['num_joints']):
-            if joints_3d_visible[i, 0] > 0.0:
-                joints_3d[i, 0:2] = affine_transform(joints_3d[i, 0:2], trans)
+        if self.use_udp:
+            trans = get_warpmatrix(r, c * 2.0, image_size - 1.0, s * 200.0)
+            img = cv2.warpAffine(
+                img.copy(),
```

Comment:
is this copy needed?

Reply:
i check this, it is not necessary, i will remove this in both top-down and bottom-up transformation